### PR TITLE
feat: use Anthropic Models API for dynamic model discovery

### PR DESCRIPTION
**Pull request by vdimarco**
## Summary
- Updated `fetch_models_from_anthropic` to use the `/v1/models` API endpoint instead of a static model list
- Added pagination support for fetching all available models
- Updated `normalize_anthropic_model` to handle the API response format (`display_name`, `created_at`)
- Added comprehensive tests for model fetching and normalization

## API Documentation
https://docs.anthropic.com/en/api/models-list

## Changes
- Removed hardcoded list of 5 Claude models
- Now dynamically fetches all available models from Anthropic's API
- Models are filtered to only include `claude-*` prefixed models
- Context length and max output are inferred from model ID patterns

## Test plan
- [x] Added unit tests for `fetch_models_from_anthropic`
- [x] Added unit tests for `normalize_anthropic_model`
- [x] Tests cover pagination, error handling, and model filtering

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h4>Greetile Summary</h4>

- Replaced hardcoded list of 5 Claude models with dynamic API-based discovery using Anthropic's `/v1/models` endpoint
- Added comprehensive test coverage for model fetching, pagination handling, error scenarios, and model normalization functionality
- Updated model normalization logic to handle API response format with `display_name` and `created_at` fields from official Anthropic API

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/services/models.py | Replaced static Claude model list with dynamic API fetching using pagination and updated normalization to handle official API response format |
| tests/services/test_anthropic_models.py | New comprehensive test file covering API fetching, pagination, error handling, and model normalization edge cases |

<h3>Confidence score: 4/5</h3>

- This PR modernizes Anthropic integration by leveraging official API instead of hardcoded models with minimal production risk
- Score reflects solid test coverage and adherence to existing patterns, with minor concerns about API dependency and error handling robustness
- Pay close attention to `src/services/models.py` for the core API integration logic and ensure the API key configuration is properly handled

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant Models Service
    participant Config
    participant HTTPX Client
    participant Anthropic API
    participant Cache
    participant Normalization
```

**Context used:**

- Context from `dashboard` - CLAUDE.md ([source](https://app.greptile.com/review/custom-context?memory=b292cd65-880f-4b4d-b2f7-a28cb17a33c4))
- Context from `dashboard` - AGENTS.md ([source](https://app.greptile.com/review/custom-context?memory=5fabd0d3-856d-4413-ab88-1b5755d16dde))

### CODE REVIEW

File: tests/services/test_anthropic_models.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    @patch("src.services.models.Config.ANTHROPIC_API_KEY", "test_key")
+    @patch("src.services.models.httpx.get")
+    def test_fetch_models_pagination(self, mock_get):
+        """Test model fetching with pagination"""
+        # Mock paginated API responses
+        page1_response = Mock()
+        page1_response.status_code = 200
+        page1_response.json.return_value = {
+            "data": [
+                {"id": "claude-3-5-sonnet-20241022", "display_name": "Claude 3.5 Sonnet", "type": "model"}
```
[the remark this answers is not in the document]
Comment (vdimarco) [reply tests/services/test_anthropic_models.py]: Fixed in commit d26dbd31 - added created_at field to pagination test data.

File: tests/services/test_anthropic_models.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
+"""Tests for Anthropic model fetching and normalization.
+
+Tests the fetch_models_from_anthropic and normalize_anthropic_model functions
+which use the Anthropic Models API: https://docs.anthropic.com/en/api/models-list
+"""
+
+import pytest
+from unittest.mock import Mock, patch, MagicMock
```
[the remark this answers is not in the document]
Comment (vdimarco) [reply tests/services/test_anthropic_models.py]: Fixed in commit d26dbd31 - removed unused MagicMock import.
